schema/entity-record.json

```diff
@@ -12,21 +12,116 @@
       "type": "boolean"
     },
     "entityType": {
+      "entityType": "object",
       "title": "Type",
-      "description": "From the entityType codelist. What kind of entity is this? The 'registeredEntity' code covers any legal entity created through an act of official registration, usually resulting in an identifier being assigned to the entity. The ‘legalEntity’ code covers other bodies with distinct legal personality (international institutions, statutory corporations etc.). The 'arrangement' code covers artificial entities, described in the data model for the purpose of associating one or more natural or legal persons together in an ownership or control relationship, but without implying that the parties to this arrangement have any other form of collective legal identity.",
-      "type": "string",
-      "enum": [
-        "registeredEntity",
-        "legalEntity",
-        "arrangement",
-        "anonymousEntity",
-        "unknownEntity",
-        "state",
-        "stateBody"
+      "description": "The type of entity described in the statement.",
+      "required": [
+        "type"
       ],
-      "codelist": "entityType.csv",
-      "openCodelist": false,
-      "propertyOrder": 4
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Entity type",
+          "description": "A value from the entityType codelist.",
+          "codelist": "entityType.csv",
+          "enum": [
+            "registeredEntity",
+            "legalEntity",
+            "arrangement",
+            "anonymousEntity",
+            "unknownEntity",
+            "state",
+            "stateBody"
+          ]
+        },
+        "subtype": {
+          "type": "string",
+          "title": "Entity subtype",
+          "description": "A value from the entitySubtype codelist. This MUST align with the `entityType` value.",
+          "codelist": "entitySubtype.csv",
+          "enum": [
+            "governmentDepartment",
+            "stateAgency",
+            "other",
+            "trust",
+            "nomination"
+          ],
+          "openCodelist": false
+        },
+        "details": {
+          "type": "string",
+          "title": "Details",
+          "description": "A local name for this type of entity, or any further information to identify the type of entity. For example, in Finland 'ministeriö' for a government department."
+        }
+      },
+      "propertyOrder": 4,
+      "if":{
+        "not":{
+          "properties":{
+            "subtype":{
+              "type":"null"
```

Comment:
I'm inferring from this that subtype could be null, but currently the schema enforces that it must be a value from the codelist - so this if statement will never evaluate to false, making it a bit redundant. Either it needs to be explicitly possible for it to be null, or you can remove a layer of nesting (and some of the complexity) here, I think.

Reply:
It can't be explicitly null but it isn't a required field. Without the 'if not null' preamble the logic breaks when you don't have an entitySubtype even thought that is allowed by the schema. 

This is a simplified schema and some examples that show what I mean 

https://docs.google.com/document/d/1vBgsjYscPtXvBhAO6k3LudqW0II7BFY3sYboMDNeXcI/edit?usp=sharing